app/main.py

```diff
@@ -1,8 +1,24 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[name] = self
 
 
 def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+    result = []
+    for person in people:
+        result.append(
+            Person(
+                name=person["name"],
+                age=person["age"]
+            )
+        )
+    for person in people:
+        if "wife" in person and person["wife"]:
+            wife = Person.people[person["wife"]]
```

Comment:
You could simplify this check by using get, e.g.
```suggestion
        if person.get("wife"):
            wife = Person.people[person["wife"]]
```
Or use a walrus operator to not retrieve the value by key twice: 
```suggestion
        if wife_name := person.get("wife"):
            wife = Person.people[wife_name]
```